miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultExperimentService.java

```diff
@@ -73,60 +73,62 @@ public NamingScheme getNamingScheme() {
   }
 
   @Override
-  public Collection<Experiment> listAllExperiments() throws IOException {
+  public Collection<Experiment> listAll() throws IOException {
     return authorizationManager.filterUnreadable(experimentStore.listAll());
   }
 
   @Override
-  public Collection<Experiment> listAllExperimentsBySearch(String query) throws IOException {
+  public Collection<Experiment> listAllBySearch(String query) throws IOException {
     return authorizationManager.filterUnreadable(experimentStore.listBySearch(query));
   }
 
   @Override
-  public Collection<Experiment> listAllExperimentsByStudyId(long studyId) throws IOException {
+  public Collection<Experiment> listAllByStudyId(long studyId) throws IOException {
     return authorizationManager.filterUnreadable(studyStore.get(studyId).getExperiments());
   }
 
   @Override
-  public Collection<Experiment> listAllExperimentsWithLimit(long limit) throws IOException {
+  public Collection<Experiment> listAllWithLimit(long limit) throws IOException {
     return authorizationManager.filterUnreadable(experimentStore.listAllWithLimit(limit));
   }
 
   @Override
-  public long saveExperiment(Experiment experiment) throws IOException {
+  public long save(Experiment experiment) throws IOException {
     try {
-    authorizationManager.throwIfNotWritable(experiment);
-    if (experiment.getId() == ExperimentImpl.UNSAVED_ID) {
+      authorizationManager.throwIfNotWritable(experiment);
+      if (experiment.getId() == ExperimentImpl.UNSAVED_ID) {
+        experimentStore.save(experiment);
```

Comment:
Will fail due to null name. Should generate a temporary name first, like is done with Samples